temporal/fx.go

```diff
@@ -956,17 +964,21 @@ var ServiceTracingModule = fx.Options(
 				if rsn == primitives.InternalFrontendService {
 					rsn = primitives.FrontendService
 				}
-				serviceName := string(rsn)
-				if !strings.HasPrefix(serviceName, "io.temporal.") {
```

Comment:
I don't see why they would unexpectedly be prefixed with `io.temporal.` - they are constants.